yt_dlp/extractor/lecturio.py

```diff
@@ -16,16 +16,27 @@
 
 class LecturioBaseIE(InfoExtractor):
     _API_BASE_URL = 'https://app.lecturio.com/api/en/latest/html5/'
+    _DE_API_BASE_URL = 'https://lecturio.de/api/de/latest/html5/'
     _LOGIN_URL = 'https://app.lecturio.com/en/login'
+    _DE_LOGIN_URL = 'https://www.lecturio.de/anmelden.html'
     _NETRC_MACHINE = 'lecturio'
 
+    is_DE = None
+
+    # Find out if url is german before starting anything else
+    def extract(self, url):
+        self.is_DE = True if re.match(r"https://(?:www\.)?lecturio\.de/", url) else False
+        return super().extract(url)
+
     def _perform_login(self, username, password):
```

Comment:
`_perform_login` is not called per-URL. You'd have to check in `real_extract` whether login is valid for current URL and call `_perform_login` as necessary

Reply:
I did is this way, because `_perform_login` is called by `initialize` , and is the first extractor specific function that is called. I don't see how I could pass additional data into extractor specific functions without using global state that is set by either extending  or overwriting `initialize`  or  `extract`